File: sykepengesoknad/src/main/java/no/nav/syfo/kafka/sykepengesoknad/dto/SykepengesoknadDTO.java
package no.nav.syfo.kafka.sykepengesoknad.dto;

import lombok.Builder;
import lombok.Value;
import no.nav.syfo.kafka.interfaces.Soknad;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Value
@Builder(toBuilder = true)
public class SykepengesoknadDTO implements Soknad {
    String id;
    SoknadstypeDTO type;
    SoknadsstatusDTO status;
    String aktorId;
    String sykmeldingId;
    ArbeidsgiverDTO arbeidsgiver;
    ArbeidssituasjonDTO arbeidssituasjon;
    String korrigerer;
    String korrigertAv;
    Boolean soktUtenlandsopphold;
    ArbeidsgiverForskuttererDTO arbeidsgiverForskutterer;
    LocalDate fom;
    LocalDate tom;
    LocalDate startSyketilfelle;
    LocalDate arbeidGjenopptatt;
    LocalDateTime sykmeldingSkrevet;
    LocalDateTime opprettet;
    LocalDateTime sendtNav;
    LocalDateTime sendtArbeidsgiver;
    List<PeriodeDTO> egenmeldinger;
    List<PeriodeDTO> papirsykmeldinger;
    List<FravarDTO> fravar;
    List<InntektskildeDTO> andreInntektskilder;
    List<SoknadsperiodeDTO> soknadsperioder;
    List<SporsmalDTO> sporsmal;
    AvsendertypeDTO avsendertype;
    boolean ettersending;
    MottakerDTO mottaker;
}

File: sykepengesoknad/src/main/kotlin/no/nav/syfo/kafka/sykepengesoknad/dto/SykepengesoknadDTO.kt
package no.nav.syfo.kafka.sykepengesoknad.dto

import no.nav.syfo.kafka.interfaces.Soknad
import java.time.LocalDate
import java.time.LocalDateTime

data class SykepengesoknadDTO(
        val id: String? = null,
        val type: SoknadstypeDTO? = null,
        val status: SoknadsstatusDTO? = null,
        val aktorId: String? = null,
        val sykmeldingId: String? = null,
        val arbeidsgiver: ArbeidsgiverDTO? = null,
        val arbeidssituasjon: ArbeidssituasjonDTO? = null,
        val korrigerer: String? = null,
        val korrigertAv: String? = null,
        val soktUtenlandsopphold: Boolean? = null,
        val arbeidsgiverForskutterer: ArbeidsgiverForskuttererDTO? = null,
        val fom: LocalDate? = null,
        val tom: LocalDate? = null,
        val startSyketilfelle: LocalDate? = null,
        val arbeidGjenopptatt: LocalDate? = null,
        val sykmeldingSkrevet: LocalDateTime? = null,
        val opprettet: LocalDateTime? = null,
        val sendtNav: LocalDateTime? = null,
        val sendtArbeidsgiver: LocalDateTime? = null,
        val egenmeldinger: List<PeriodeDTO>? = null,
        val papirsykmeldinger: List<PeriodeDTO>? = null,
        val fravar: List<FravarDTO>? = null,
        val andreInntektskilder: List<InntektskildeDTO>? = null,
        val soknadsperioder: List<SoknadsperiodeDTO>? = null,
        val sporsmal: List<SporsmalDTO>? = null,
        val avsendertype: AvsendertypeDTO? = null,
        val ettersending: Boolean = false,
        val mottaker: MottakerDTO? = null,
        val fnr: String? = null
) : Soknad
